hubploy/auth.py

```diff
@@ -216,32 +232,45 @@ def cluster_auth_gcloud(deployment, project, cluster, zone, service_key):
     yield
 
 
-def cluster_auth_aws(deployment, project, cluster, zone, service_key):
+def cluster_auth_aws(deployment, project, cluster, zone, service_key = None, eks_role = None):
     """
     Setup AWS authentication with service_key
 
     This changes *global machine state* on what current kubernetes cluster is!
     """
-    # Get credentials from standard location
-    service_key_path = os.path.join(
-        'deployments', deployment, 'secrets', service_key
-    )
 
-    original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
+    if not service_key and not eks_role:
+        raise Exception('AWS authentication requires either a service key or the use of a role')
 
-    try:
+    if service_key:
+        # Get credentials from standard location
+        service_key_path = os.path.join(
+            'deployments', deployment, 'secrets', service_key
+        )
 
-        # Set env variable for credential file location
-        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
+        original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
 
-        subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
-                               '--name', cluster, '--region', zone])
+        try:
+            # Set env variable for credential file location
+            os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
 
-        yield
+            subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
```

Comment:
This needs to happen for both role assumption and for user accounts. This generates the KUBECONFIG file that tells `kubectl` to use aws for authentication. Then aws can decide to use credential files or roles, but we still need to tell kubectl.